### PR TITLE
sqlite3 use different name for bin target to avoid collision in pdb file names

### PR DESCRIPTION
**Pull request by boussaffawalid**
**sqlite3/3.36.0**

Using the same target name for the library and the executable will lead to collision in pdb file names on windows.

sqlite3.exe will generate sqlite3.pdb
sqlite3.dll will generate sqlite3.pdb

pdb files override each other.
 

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/sqlite3/all/CMakeLists.txt [hunk 1]
```diff
@@ -150,7 +150,6 @@ install(DIRECTORY source_subfolder/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
 if(SQLITE3_BUILD_EXECUTABLE)
     add_executable(sqlite3-bin source_subfolder/shell.c)
     target_link_libraries(sqlite3-bin PRIVATE ${PROJECT_NAME})
-    set_target_properties(sqlite3-bin PROPERTIES OUTPUT_NAME "sqlite3")
```
Comment (SpaceIm) [inline recipes/sqlite3/all/CMakeLists.txt:153]: don't do that, sqlite3 executable name must be sqlite3.

File: recipes/sqlite3/all/CMakeLists.txt [hunk 2]
```diff
@@ -150,7 +150,6 @@ install(DIRECTORY source_subfolder/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
 if(SQLITE3_BUILD_EXECUTABLE)
     add_executable(sqlite3-bin source_subfolder/shell.c)
     target_link_libraries(sqlite3-bin PRIVATE ${PROJECT_NAME})
```
Comment (SpaceIm) [inline recipes/sqlite3/all/CMakeLists.txt:152]:
```suggestion
    target_link_libraries(sqlite3-bin PRIVATE ${PROJECT_NAME})
    set_target_properties(sqlite3-bin PROPERTIES OUTPUT_NAME "sqlite3" PDB_NAME "sqlite3-bin")
```

**Comment by SpaceIm** [comment]
Could you add sqlite3 to the title please?

**Review by madebr — APPROVED** [review]

File: recipes/sqlite3/all/CMakeLists.txt (hunk 1 above)
Comment (uilianries) [reply recipes/sqlite3/all/CMakeLists.txt:153]: Agreed, that's the original name, you must not change it.

**Review by uilianries — CHANGES_REQUESTED** [review]
Lock to prevent merging. Need another alternative, removing the original name is not a good option.

File: recipes/sqlite3/all/CMakeLists.txt (hunk 1 above)
Comment (ericLemanissier) [reply recipes/sqlite3/all/CMakeLists.txt:153]: I think this was already addressed in https://github.com/conan-io/conan-center-index/pull/8586/commits/3170293346757f6452044fe30523807f5af11c2e

Comment (uilianries) [reply recipes/sqlite3/all/CMakeLists.txt:153]: Right, removing the block.

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]
The suggestion were applied as far as I can tell. I think this is good to merge

**Comment by prince-chrismc** [comment]
You'll need to retrigger CI, close the pr wait 10s and then re-open it 🔁

**Review by uilianries — APPROVED** [review]

**Pull request merged**